free-3.md

```diff
@@ -7,44 +7,126 @@ good way to meet their needs.
+By creating a formal markup, we could better summarize behaviors of students and
+graders.
 
 ### Why a language?
 _Why is a DSL appropriate for your user(s)? How does it address the need?_
 
+Graders in CS are usually pretty comfortable with programming, so a language
+would fit in. Furthermore, creating a formal DSL will require the graders to
+be more precise in their feedback and allow professors to analyze larger grading
+trends.
+
+Having the feedback in a machine readable form would allow us to present it to
+users in different ways and make it searchable.
 
 ### Why you?
 _What excites you about this idea? How did you come up with it?_
 
+Last semester I created an automated grading system that integrated testing
+results with point assigned rubrics to automatically populate an electronic
+rubric that TAs could comment and override. This system could only support
+key value pairs and specific code comments were left on PDFs later on which felt
+redundent.
 
 ### Domain
 _Describe the project's domain in five words._
 
+Efficient code feedback and grading.
 
 ### Interface (syntax)
-_How might the user interact with the language? What does programming look 
-like? Why is this the right way to interact with the problem domain?_ 
+_How might the user interact with the language? What does programming look
+like? Why is this the right way to interact with the problem domain?_
+
+I imagine a program to look something like this:
+
+```
+include: common
+assignment: "Hello, World"
+feedback:
+  style::category:
+    -1 @ {1:56}: :common.problems.variables.names
+    -0.5 @ {47:*-50:*}: :common.problems.semicolons
+    -2 @ {62:*}: custom.("Custom deduction method comment")
+```
+
+that produces output like:
+
+```
+1> int a_var = 123123120931923012;
+
+1 point was deducted based on the above line.
+Please make sure you choose names for your variables with meaning.
 
+47> public class MyObject {
+48>    public MyObject(int x, int y) { super();;;;; }
+49>    public void get() { return this;;;;;;; }
+50> }
+
+0.5 points were deducted based on the above lines.
+Please make sure you only use one semicolon.
+
+62> int myInt = 0;
+
+2 points were deducted based on the above lines.
+Custom deduction method comment.
+```
+
+where you can specify categories and then specific line numbers that describe
+the deduction (or simply leave a comment).
+
+I think this make sense since it is easy to read and efficient to write. It could
+be presented alongside the document in the appropriate place making it more
+readable.
+
+Defining a grading "grammar" that can be course specifc with namespaced shortcuts
+for longer comments creates a more efficient grading experience. It also allows
+us to analyze trends. If we notice a bunch of `common.problems.semicolons`, we
+can look into it further. Without a DSL, we have don't have an easy way to notice
+trends across graders and students in larger courses.
 
```

Comment:
I'm not fully convinced that typing common.problems.variables.names is a significant timesaver. That being said, when grading 81, I found myself copy-pasting "See sample solution" an inordinate amount of time.